File: app/src/main/java/com/asmat/rolando/popularmovies/ui/fragments/FavoriteMoviesGridFragment.java
package com.asmat.rolando.popularmovies.ui.fragments;

import android.arch.lifecycle.LiveData;

import com.asmat.rolando.popularmovies.ui.adapters.grid.FavoriteMoviesGridAdapter;
import com.asmat.rolando.popularmovies.ui.adapters.grid.MoviesGridBaseAdapter;
import com.asmat.rolando.popularmovies.database.DatabaseManager;
import com.asmat.rolando.popularmovies.ui.adapters.MovieAdapterOnClickHandler;

public class FavoriteMoviesGridFragment extends BaseGridFragment {

    public FavoriteMoviesGridFragment() { }

    @Override
    MoviesGridBaseAdapter getAdapter(MovieAdapterOnClickHandler handler) {
        return new FavoriteMoviesGridAdapter(handler);
    }

    @Override
    LiveData<Movie[]> getMovieSource() {
        return DatabaseManager.INSTANCE.getFavoriteMovies();
    }

}

File: app/src/main/java/com/asmat/rolando/popularmovies/ui/fragments/FavoriteMoviesGridFragment.kt
package com.asmat.rolando.popularmovies.ui.fragments

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.Transformations
import com.asmat.rolando.popularmovies.ui.mylists.FavoriteMoviesGridAdapter
import com.asmat.rolando.popularmovies.model.Movie
import com.asmat.rolando.popularmovies.model.mappers.MovieMapper
import com.asmat.rolando.popularmovies.ui.common.MovieAdapterOnClickHandler
import com.asmat.rolando.popularmovies.ui.common.BaseGridFragment
import com.asmat.rolando.popularmovies.ui.common.BaseMoviesGridAdapter

class FavoriteMoviesGridFragment : BaseGridFragment() {

    override val movieSource: LiveData<List<Movie>>
        get() {
            val source = moviesRepository.getAllFavoriteMovies()
            return Transformations.map(source) { favoriteMovies ->
                favoriteMovies.map {
                    MovieMapper.from(it)
                }
            }
        }

    override fun getAdapter(handler: MovieAdapterOnClickHandler): BaseMoviesGridAdapter {
        return FavoriteMoviesGridAdapter(handler)
    }

}